Update paragraph tables with references to new field names

andybroomfield opened the pull request:
Fix #75

If the field storage is a paragraph, an extra step is executed:
  The tables `paragraphs_item_field_data` and
  `paragraphs_item_revision_field_data` are updated for the entity type
  and source field name to the new field name.

eg. updated parent_type = 'node' and parent_field_name = 'field_topics'
is updated to parent_field_name = 'localgov_topics'

andybroomfield committed:
Update paragraph tables with references to new field names

Fix #75

If the field storage is a paragraph, an extra step is executed:
  The tables `paragraphs_item_field_data` and
  `paragraphs_item_revision_field_data` are updated for the entity type
  and source field name to the new field name.

eg. updated parent_type = 'node' and parent_field_name = 'field_topics'
is updated to parent_field_name = 'localgov_topics'

andybroomfield committed:
Coding standards fixes

andybroomfield commented:
Will also need to be ported to 2.x branch?

andybroomfield commented:
First pass @Adnan-cds Let me know if this works for you.

Adnan-cds commented:
Hi Andy,
The parent field name has been correctly changed to **localgov_topics** in my testing.  I have deployed this change to one of our dev sites with an outdated database where field renaming has not happened yet.  That site looks okay just after deployment.  I will confirm tomorrow if it is still unaffected.

Adnan-cds commented:
> I will confirm tomorrow if it is still unaffected.

Dev site looks good.  Fix seem to be working :)

andybroomfield commented:
Next step is to get an automated test written for this, if anyone has tips on creating a test with paragraph fields.

andybroomfield committed:
Add test for paragraph fix on field rename

- Create a test to check the paragraph fields remain intact post cron runs
- Adds paragraphs as a test dependency

andybroomfield committed:
Coding standards fix and rename the test

andybroomfield commented:
Putting this back in draft as the tests are passing even if the paragraph fix is removed.
... So its finding out what is causing the underlying deletion and running that directly in the test, as just running cron wont work.

andybroomfield committed:
Fix test to call the entity reference revisions purger

The missing function call to remove the orphaned paragraphs is
_entity_reference_revisions_orphan_purger_batch_dispatcher

andybroomfield committed:
Add extra if to fieldrenamehelper to detect entity_referenece_revisions

TODO: add logger that non paragraph fields are not currently supported

andybroomfield commented:
I get a test to work (and fail by doing a different type) by running the private function from entity referenece revisions: 
````php
_entity_reference_revisions_orphan_purger_batch_dispatcher('entity_reference_revisions.orphan_purger:deleteOrphansBatchOperation', 'paragraph', []);
````
As the current problem is just paragraphs this fix will work for that.
However all entity referenece revisions fields will also need this fix.
We could just set an exception now if another field type attempts to use this, just saying only paragraphs are supported. 
Really this class should only be called from our update hooks anyway.

Adnan-cds commented:
> We could just set an exception now if another field type attempts to use this, just saying only paragraphs are supported.

Fine by me :)

andybroomfield committed:
Throw exception if renaming a non paragraph entity_reference_revisions

andybroomfield commented:
This is ready for review now @Adnan-cds @ekes

Adnan-cds reviewed (approved):
I have finished testing these latest changes on an older copy of the Croydon site where fields were still using the "field_" prefix.  Site looks good.  Tests are also passing.  Code is impressive.  Thank you.

andybroomfield commented:
I'm going to merge this, though #78 might need to be looked at if its a serious issue.
